pybossa/view/projects.py

```diff
@@ -3032,10 +3032,21 @@ def transfer_ownership(short_name):
     if request.method == 'POST' and form.validate():
         new_owner = user_repo.filter_by(email_addr=form.email_addr.data)
         if len(new_owner) == 1:
-            new_owner = new_owner[0]
-            project.owner_id = new_owner.id
-            project.owners_ids = [new_owner.id]
+            old_owner_id = project.owner_id
+            new_owner_id = new_owner[0].id
+
+            # Assign new project owner id.
+            project.owner_id = new_owner_id
+
+            # Remove old project owner id from coowners.
+            project.owners_ids.remove(old_owner_id)
+
+            # Add new project owner id to coowners.
+            project.owners_ids.append(new_owner_id)
```

Comment:
Perhaps we can check whether `new_owner_id` already exists in `project.owner_ids` before appending it to avoid duplicate?

Reply:
Great suggestion! I've added a unit test to cover this as well. Thank you!
https://github.com/bloomberg/pybossa/pull/726/commits/bcee88822411565c58d19eb63757303b11d65c9d